File: BackendKaikan/src/main/java/com/kaikan/model/plato/Plato.java
package com.kaikan.model.plato;

import com.kaikan.generics.BaseEntity;
import com.kaikan.model.categoria.Categoria;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table (name = "plato", schema = "public")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Plato extends BaseEntity {
    @NotNull
    private String nombre;
    private String descripcion;
    @NotNull
    @PositiveOrZero
    private Double precio;
    private String img_url;
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn (name = "id_categoria", nullable = false)
    @NotNull
    private Categoria categoria;
}

File: Frontend-Kaikan-Restobar/src/app/core/models/plato.ts
import { Categoria } from "./categoria";

export class Plato {
    id: number | null;
    nombre: string;
    descripcion: string;
    img_url: string;
    precio: number;
    categoria: Categoria;

    constructor( id:number | null, nombre: string, descripcion: string, img_src: string, precio: number, categoria: Categoria ){
        this.id = id;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.img_url = img_src;
        this.precio = precio;
        this.categoria = categoria;
    }

}